Translation Linting

akatov opened the pull request:
fixes #17614

### Aims

We want to lint translation usage in the app, in particular we want to know about
1. *undefined translation keys* (e.g. usage of `:t/some-key` inside `src`, where `"some-key"` is not defined in `translations/en.json`)
2. *non-namespaced translation keys* (e.g. usage of `some-other-key` inside `src`, where `"some-other-key"` is defined in `translations/en.json`)
3. *unused translation keys* (e.g. presence of `"unused-key"` in `translations/en.json`, where `:t/unused-key` is never used inside `src`.

Some of this (2. and 3.) was already done by `test/appium/tests/non_func/test_translations.py` but in a naive way, basically simply doing a full text search of translation keys inside `src`. It also highlighted unused translation keys that end in a number as "to be rechecked".

### Summary

This PR adds bab a [babashka](https://babashka.org/) and a babashka script `scripts/lint_translations.clj`, that uses `clj-kondo` to analyze `src` as clojure source code rather than pure text. We thus discover quite a few false negatives that were missed by the python script as it e.g. picked up non-namespaced keywords and even strings that matched a translation key.

We don't distinguish between 2. and 3. (from above) as there would be too many false positives - for instance, there is a `"bold"` translation key in `translations/en.json`, and the non-namespaced `:bold` keyword is present many times in `src`, however, most of its occurrences are not used for translation. We may want to revisit this distinction again once all actual translation keys in the code are namespaced, to catch future non-namespaced translation keys. In that case, check out the last commit in this PR.

We include translation linting as part of `make lint`.

We provide a translation for `goerli-network` and we fix the mis-spelled translation key `ens-username-continue` to `ens-username-owned-continue` to make `make lint` pass.

Finally, we remove the old translation linting python script as its functionality is now superseded by the babashka script.

akatov committed:
clj-kondo translation checker

akatov committed:
clean up

akatov committed:
return status of translation linting script

akatov committed:
better name for lint-translations script

akatov committed:
cleanup

akatov committed:
better name for clj-kondo hook

akatov committed:
turn lint_translation.clj into a babashka script

akatov committed:
run clj-kondo translation linting

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 5728421a | [#1](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-17735/1/) | 2023-10-31 13:23:48 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-17735/1/consoleText) |

akatov committed:
only check for missing translation keys

akatov committed:
don't change the structure of i18n/label nodes for linting!

akatov committed:
more determinism when registering translation keywords

akatov committed:
exit lint script with proper return value

akatov committed:
also lint possibly missing translations

do not fail the script though as there are currently too many of them to be fixed in one go

akatov committed:
no need to registered translation keyword usage by clj-kondo as we are now filtering on keywords themsleves

akatov committed:
sort the possibly unused translation keys since translations/en.json is sorted

akatov committed:
remove translation linting python script

akatov commented:
Here are the false negatives that were not picked up by the translation linting python script:

```
NFT
about-key-storage-content
about-key-storage-title
about-names-content
about-names-title
access-key
account-added
accounts
active-members
active-online
active-unknown
add-an-account
add-contact
add-nickname
add-text
added
advanced-settings
agree-by-continuing
app-connections
apr
ask-in-status
audio-recorder
aug
available
available-participants
back-up
balance
biometric-auth-reason-verify
biometric-enable
biu
block-contact
bold
browsed-websites
browsers
cancel-keycard-setup
cannot-read-card
card-is-blank
card-reseted
category
category-title
change-logging-enabled
change-passcode
change-password
changed-amount-warning
changed-asset-warning
chaos-mode
chaos-unicorn-day
chaos-unicorn-day-details
chat-name
chat-settings
chats
clear-all
clear-history-confirmation
close-app-content
command-button-send
community-color
community-info
community-link
complete-hardwallet-setup
completed
confirm-new-password
confirmations-helper-text
connecting-requires-login
connection-with-the-card-lost
connection-with-the-card-lost-setup-text
connection-with-the-card-lost-text
contact-request-sent-to
contact-s
contacts-descr
copy-info
copy-to-clipboard
create-category
create-channel
create-community
create-multiaccount
create-pin
create-pin-description
created-group-chat-description
cryptokitty-name
currency
current
current-average-tip
current-base
current-network
current-password
cut
datetime-day
datetime-hour
datetime-minute
datetime-second
dec
decryption-failed-content
default
delete-and-leave-group
delete-message
delete-my-account
description
dev-mode
dev-mode-settings
disable
disabled
disconnected
edit-community
edit-group
edit-name-image
edit-nickname
edit-profile
empty-chat-description
empty-chat-description-one-to-one
empty-chat-description-public
empty-chat-description-public-share-this
encrypt-with-password
ens-primary-username
ens-remove-hints
ens-show-username
ens-test-message
enter-12-words
enter-pin
enter-seed-phrase
error-unable-to-get-balance
error-unable-to-get-prices
error-unable-to-get-token-balance
errors
eth
ethereum-node-started-incorrectly-description
ethereum-node-started-incorrectly-title
export-account
export-key
favourite
feb
fetch-community
fetching-community
find
finishing-card-setup
follow
fr
fri
generate-a-key
generate-a-new-key
generate-account
generating-codes-for-pairing
generating-mnemonic
get-status-at
give-a-short-description
go-to-settings
got-it
group-chat
group-chat-admin-added
group-chat-created
group-chat-member-added
group-chat-member-joined
group-chat-member-removed
group-chat-name-changed
help
help-center
hold-card
home
hooks
identifier
image-remove-current
image-source-gallery
image-source-make-photo
image-source-title
import-community
in-contacts
include
incorrect-code
initialization
intro-message1
intro-privacy-policy-note1
intro-privacy-policy-note2
intro-text
intro-text1
intro-text2
intro-text3
intro-title1
intro-title2
intro-title3
intro-wizard-text1
intro-wizard-text2
intro-wizard-text3
intro-wizard-text6
intro-wizard-title-alt4
intro-wizard-title-alt5
intro-wizard-title1
intro-wizard-title2
intro-wizard-title3
invalid-key-confirm
invalid-key-content
invalid-range
invite-contacts
invited
italic
jan
join
join-group-chat
join-group-chat-description
joined-group-chat-description
jul
jun
keycard-desc
keycard-onboarding-finishing-header
keycard-onboarding-pairing-header
keycard-onboarding-preparing-header
keycard-onboarding-start-header
keycard-onboarding-start-step1
keycard-onboarding-start-step1-text
keycard-onboarding-start-step2
keycard-onboarding-start-step2-text
keycard-onboarding-start-step3
keycard-onboarding-start-step3-text
keycard-onboarding-start-text
language
left
les-ulc
loading
logging
logging-enabled
lower-than-average-tip
mailserver-connection-error
mailserver-reconnect
mailserver-request-error-status
mar
max-fee
may
members
members-active
members-active-none
message-not-sent
message-options-cancel
might-break
migrations-failed-content
mo
mobile-network-continue-syncing
mobile-network-sheet-configure
mobile-network-sheet-remember-choice
mobile-network-sheet-settings
mobile-network-stop-syncing
mobile-network-stop-syncing-details
mobile-syncing-sheet-details
mobile-syncing-sheet-title
mon
multiaccounts-recover-enter-phrase-text
mute-chat
mute-group
my-profile
my-status
network-invalid-network-id
network-invalid-status-code
network-invalid-url
new-contact
new-group
new-password
new-public-group-chat
new-status
new-ui
no-pairing-slots-available
no-result
no-tokens-found
node-info
non-contacts
not-found
notify
nov
oct
off
open-chat
open-dapp
open-home
optional
pair-devices
pairing
pairing-card
passphrase
password-placeholder
password-placeholder2
password_error1
paste-json
pay-to-chat
pending-confirmation
pending-invitations
phone-e164
photos
pin
pin-code
pinned-by
preview-privacy
product-information
profile-details
public-chat
public-chats
public-group-status
public-group-topic
public-key
puk-and-pairing-codes-displayed
quiet-days
quiet-hours
re-encrypt-key
receive-transaction
received
recipient-code
recover
recover-keycard-multiaccount-not-supported
recover-with-keycard
recovering-key
recovery-confirm-phrase
recovery-success-text
recovery-typo-dialog-description
recovery-typo-dialog-title
refresh
registered
remember-me
remove-group
remove-network
removed
request-access
request-membership
request-pending
requesting
required-field
resend-message
reset-database-warning
sa
sat
save-password
save-password-unavailable
save-password-unavailable-android
see-details
see-suggestions
select
select-chat
selected
send-message
send-request
send-request-amount-max-decimals
send-request-unknown-token
send-sending-to
sent-at
sep
server
share-chat
share-contact-code
share-image
share-my-profile
share-profile-link
share-public-chat-text
shared
sharing-copy-to-clipboard
sharing-share
show-less
show-more
sign transactions
sign-message
sign-you-in
signing
simple
skip
soon
start-chat
start-conversation
start-group-chat
start-new-chat
status
status-hardwallet
status-not-sent-click
status-not-sent-tap
status-pending
status-sent
status-tx-not-found
strikethrough
su
submit
success
sun
sync-code-generated
sync-in-progress
sync-synced
tag-was-lost
text-input-disabled
th
this-device-desc
this-will-take-few-seconds
thu
timeline
to-block
to-encrypt-enter-password
to-see-this-message
token-auto-validate-decimals-error
token-auto-validate-name-error
token-auto-validate-symbol-error
token-price
topic-name-error
transaction
transaction-declined
transaction-history
transaction-request
transactions-filter-select-all
transactions-filter-title
transactions-history
transactions-history-empty
transactions-history-loading
tribute-required-by-multiaccount
tribute-state-paid
tribute-state-pending
tribute-state-required
tribute-to-talk
tribute-to-talk-add-friends
tribute-to-talk-are-you-friends
tribute-to-talk-ask-to-be-added
tribute-to-talk-contact-received-your-tribute
tribute-to-talk-desc
tribute-to-talk-disabled
tribute-to-talk-disabled-note
tribute-to-talk-enabled
tribute-to-talk-finish-desc
tribute-to-talk-learn-more-1
tribute-to-talk-learn-more-2
tribute-to-talk-learn-more-3
tribute-to-talk-paywall-learn-more-1
tribute-to-talk-paywall-learn-more-2
tribute-to-talk-paywall-learn-more-3
tribute-to-talk-pending
tribute-to-talk-pending-note
tribute-to-talk-removing-note
tribute-to-talk-set-snt-amount
tribute-to-talk-signing
tribute-to-talk-transaction-failed-note
tribute-to-talk-tribute-received1
tribute-to-talk-tribute-received2
tribute-to-talk-you-require-snt
tu
tue
ulc-enabled
unblock-contact
unknown-status-go-error
unlimited
unmute-chat
unmute-group
unpin
until
url
usd-currency
use-valid-contact-code
view-cryptokitties
view-cryptostrikers
view-data
view-etheremon
view-gitcoin
view-rules
view-superrare
waiting-for-wifi
waiting-for-wifi-change
waiting-to-sign
wallet-connect
wallet-settings
warning-message
we
web-view-error
wed
welcome-screen-text
welcome-to-status
welcome-to-status-description
word-count
write-down-and-store-securely
wrong-contract
you-can-change-account
your-data-belongs-to-you
your-data-belongs-to-you-description
your-keys
```

akatov committed:
even better location for status-im clj-kondo hooks

akatov committed:
make lint-fix

akatov committed:
report missing translations (which make the lint-translations script fail) towards the end

this way the developer is more likely to notice them and fix them first

akatov committed:
run translation linting as part of `make lint`

akatov committed:
alphabetical ordering of "ens-username-*" English translation keys

akatov committed:
add English translation for goerli-network

akatov committed:
fix to most likely intended translation keyword

akatov committed:
add babashka to nix

akatov committed:
clean up script a bit

J-Son89 commented on scripts/lint_translations.clj:
```diff
@@ -0,0 +1,77 @@
+(require '[cheshire.core :as json])
+(require '[clojure.set :as set])
+
+(defn- analyze-code
```
should these helpers be moved to a more generic or broader in scope ns?

akatov replied:
Re `analyze-code` - I think for now it's probably easier just to use `kondo/run!` inside `-main` directly.

Regarding `safe-name` and `->keyword` - they are quite specific to this use case (analysing all keywords in the source code) and I'd argue that it is too early to move them to a separate namespace.

`report-issues` is possibly the one function that should be extracted, but I'd argue that, again, it is too early, as we don't know for sure exactly how we want to report problems in the future (e.g. whether we want to include the namespace and the column besides the filename and the row, etc.).

J-Son89 commented on scripts/lint_translations.clj:
```diff
@@ -0,0 +1,77 @@
+
+(def ^:private translation-file "translations/en.json")
+
+;; (def unused-warning (format "Unused Translation Key in %s:" translation-file))
```
can we delete commented code instead 🙏

akatov replied:
hmm, all the comments here were left half intentionally for the case we ever want to distinguish between unused translations and usage of non-namespaced translation keys...

As I pointed out in the description this is not always generally possible (as there would be too many false postivies for non-namespaced translation keys), and certainly not appropriate at this stage, but could be appropriate in the future once all actual translation keys in the code are namespaced, to catch future non-namespaced translation keys.

akatov replied:
I am happy to delete the commented out code for now though, as it should fairly straightforward to re-create this functionality in the future if and when needed.

J-Son89 replied:
yeah, in my perspective I think it's better to have this tracked in a separate issue or in this pr description/self comments so that if we ever want to revisit we can return to the pr and see any extra details that might be relevant about this work.

akatov replied:
updated description

J-Son89 commented on scripts/lint_translations.clj:
```diff
@@ -0,0 +1,77 @@
+        used-translation-keys            (set (map :name used-translations))
+        possibly-unused-translation-keys (set/difference file-translation-keys used-translation-keys)
+
+        ;;
```
same here, let's delete 👍

J-Son89 commented on scripts/lint_translations.clj:
```diff
@@ -0,0 +1,77 @@
+
+(defn -main
+  [& _args]
+  (let [result                           (analyze-code ["src"])
```
should we put "src" in a named def at the top of the file ?

J-Son89 commented on scripts/lint_translations.clj:
```diff
@@ -0,0 +1,77 @@
+  [file]
+  (-> file slurp json/parse-string keys))
+
+(def ^:private translation-file "translations/en.json")
```
should we put "translations/en.json" in a named def at the top of the file?

akatov replied:
will move it next to `src-paths`.

akatov committed:
applying review suggestions

J-Son89 reviewed (approved):
nice work! :)

flexsurfer commented:
hey @akatov thanks a lot for your work, does this script check outdated keys in en.json? because there should be many

ilmotta commented on .clj-kondo/status-im/utils/i18n.clj:
```diff
@@ -0,0 +1,26 @@
+(ns utils.i18n
```
Looking at the namespace change now @akatov, and the changes don't make much sense to me. Utility namespaces are supposed to mean they can be called/reused to build something else.

Originally, I added the `label` hook to a `core` namespace because this is part of our public hooks API so to speak. No other code should call `label`, except clj-kondo itself.

Could you please revert this namespace change?

akatov replied:
The intention was to match the namespace and name of the symbols that are supposed to be linted, in this case `utils.i18n/label`, which happens to live in the `utils` namespace.

This way, the `config.edn` could look like this in the future:
```
{:hooks {:analyze-call {utils.i18n/label utils.i18n/label
                        a.b.c/x          a.b.c/x
                        ...
                        }}
 ...
 }
```

This was namespace change was in part inspired by the same approach in e.g.
- https://github.com/taoensso/encore/blob/master/resources/clj-kondo.exports/taoensso/encore/config.edn and 
- https://github.com/taoensso/encore/blob/master/resources/clj-kondo.exports/taoensso/encore/taoensso/encore.clj

This was NOT supposed to be a utility namespace, but I see the confusion.

I am happy to change this back, as this is the only hook so far, and the above approach can be discussed once more hooks are needed.

ilmotta replied:
Yeah, hooks are structured in different ways, it's fine if you want to keep it. I guess we have so little usage of hooks at the moment we don't need to overthink.

akatov commented:
> hey @akatov thanks a lot for your work, does this script check outdated keys in en.json? because there should be many

Yes it does. It prints them to STDOUT, but it doesn't fail, since a) there are so many of them, and b) I decided not to distinguish between whether a key is unused or whether it was used in `src` without the `t:/` namespace. See the PR description summary or [this comment](https://github.com/status-im/status-mobile/pull/17735#discussion_r1380565669) for more details.

There are currently 752 potentially unused translation keys, however most of them are most likely used without the `t:/` keyword.

Another reason that I decided not to print out "definitely unused translation keys" is that in theory we can never check that for sure, as a key could be created dynamically at run-time.

However, thinking about this, I could probably restore that distinction, i.e. split the list of "Possibly Unused Translation Keys" into two lists of "Most Likely Unused Translation Keys" and "Possible Non-namespaced translation keys" (the latter with code locations). I could even make the script fail if the list of "Most Likely Unused Translation Keys" is non-empty and relatively short, and then remove all the corresponding entries from `translations/*.json`? What do you think, @flexsurfer ?

ilmotta commented on Makefile:
```diff
@@ -316,7 +316,8 @@ lint: ##@test Run code style checks
 	sh scripts/lint-trailing-newline.sh && \
-	node_modules/.bin/prettier --write .
+	node_modules/.bin/prettier --write . && \
+	scripts/lint_translations.clj
```
The `lint` target is quite slow, so I think the most valuable checks should be performed earlier. I think this linting step should happen after `clj-kondo` is called. The formatting checks are less valuable and for many unnecessary (usually) because we format automatically in the editor when saving files.

ilmotta commented on scripts/lint_translations.clj:
```diff
@@ -0,0 +1,59 @@
+  (-> file slurp json/parse-string keys))
+
+
+(def ^:private possibly-unused-warning
```
I've got tons of messages "Possibly Unused ...", which kind of pollutes the output from `make lint` substantially.

Also, some (I haven't checked all) are really not used, like `whats-on-your-mind`, but we're flagging this translation as *possibly unused*.

Could you explain why this is happening? And if we can fix?

akatov replied:
Please see my [earlier comment](https://github.com/status-im/status-mobile/pull/17735#issuecomment-1792343874) about this.

flexsurfer commented:
> without the `t:/` namespace

hm i thought it was fixed by @ilmotta here https://github.com/status-im/status-mobile/pull/17610

flexsurfer commented:
actually what im thinking, we removed a lot of UI recently, so probably expected that we have lots of unused keys in the en.json, so i would be great if we address this in this PR, removed them , fix them if there is wrong usage in the code

akatov commented:
> > without the `t:/` namespace
> 
> hm i thought it was fixed by @ilmotta here #17610

This only caught keywords that were used directly inside a call to `i18n/label`, AND outside the now deprecated `utils.re-frame/defn` macro!

There are still a lot of keywords defined elsewhere that are passed through to `i18n/label` later through some intermediate functions, and they don't get caught by the linter.

akatov commented:
There are currently 364 probably unused translation keys, and 388 probably non-namespaced translation key usages!
This is a lot of manual work...

ilmotta commented:
@akatov, if we step back a little bit, we need a script that's more precise than the Python one and not too verbose. Also the complexity of the clj script needs to be justified because it's not necessarily trivial.

If the linter output is too verbose due to the impossibility of checking dynamic values, then I can imagine someone will try to disable or move the check away from `make lint` to keep the lint noise in check.

It's starting to grow on me the idea of using something much simpler, like a `:pre` condition on the `i18n/label` function or instrumenting it with malli after it gets in the repo, which should happen sometime next week.

But I still believe the script can be valuable, just let's keep an eye on the idea that we can still go back to a simpler solution, albeit not a static check.

ilmotta commented:
> There are currently 364 probably unused translation keys, and 388 probably non-namespaced translation key usages!
> This is a lot of manual work...

Yes, if you think it's too much manual work, then I think it may be too much for a single PR to do. If the script gets merged, then a follow-up PR could fix the rest.

J-Son89 commented:
I believe the manual work requested here is beyond the scope of this issue/pr and we should track that in a separate issue.

akatov commented:
> I believe the manual work requested here is beyond the scope of this issue/pr and we should track that in a separate issue.

@J-Son89 , @flexsurfer , @ilmotta - I created #17811 and #17813 to address the proper namespacing of all used translation keywords, and the removal of unused keys, respectively.

akatov commented:
> If the linter output is too verbose due to the impossibility of checking dynamic values, then I can imagine someone will try to disable or move the check away from `make lint` to keep the lint noise in check.

The script's verbosity is not due to the impossibility of checking dynamic values, it is currently due to sheer amount of unused / non-namespaced keywords. I am commenting out the lines that actually produce the verbose output and make the script fail, until we address them in #17811  and #17813. I believe it still makes sense to keep the verbose output to catch future errors which should be relatively few from then on.

> It's starting to grow on me the idea of using something much simpler, like a `:pre` condition on the `i18n/label` function or instrumenting it with malli after it gets in the repo, which should happen sometime next week.
> 
> But I still believe the script can be valuable, just let's keep an eye on the idea that we can still go back to a simpler solution, albeit not a static check.

I think both can work together to address different kinds of mis-uses of `i18n/label`.

akatov committed:
only print out missing translations for now, but add back code to print out (probably) unused and non-namespaced translation keys to be used to remove/fix them in the future

akatov committed:
run translation linting earlier in the linting pipeline

ilmotta commented:
> > If the linter output is too verbose due to the impossibility of checking dynamic values, then I can imagine someone will try to disable or move the check away from `make lint` to keep the lint noise in check.
> 
> The script's verbosity is not due to the impossibility of checking dynamic values, it is currently due to sheer amount of unused / non-namespaced keywords. I am commenting out the lines that actually produce the verbose output and make the script fail, until we address them in #17811 and #17813. I believe it still makes sense to keep the verbose output to catch future errors which should be relatively few from then on.
> 
> > It's starting to grow on me the idea of using something much simpler, like a `:pre` condition on the `i18n/label` function or instrumenting it with malli after it gets in the repo, which should happen sometime next week.
> > But I still believe the script can be valuable, just let's keep an eye on the idea that we can still go back to a simpler solution, albeit not a static check.
> 
> I think both can work together to address different kinds of mis-uses of `i18n/label`.

Thanks @akatov for the explanation and for creating the issues.

> I think both can work together to address different kinds of mis-uses of `i18n/label`.

Agreed :)

I tried the script one last time today (because I have to leave now), and I enabled all outputs to check. The messages `Probably non-namespaced Translation Key` gave literally thousands of warnings for keywords that have no relationship with `i18n/label`. I believe this should be fixed for the script to be ready for merge or the part of the code that produces that output removed because it's unusable as it is now.

On the other hand, `Probably unused translation key` messages are much better, those really seem to not be used as far as I tested and are actually useful.

akatov commented:
@ilmotta,

> The messages `Probably non-namespaced Translation Key` gave literally thousands of warnings for keywords that have no relationship with `i18n/label`.

These are the *individual* occurrences of keywords in `src` that match a translation key in `translations/en.json`, and for which that translation key is never used in `src` with a `t:/` prefix. There are indeed currently 3153 of them (occurrences). Most of them are false positives, i.e. very common keywords like `:bold` or `:default` that just also happen to correspond to a translation key. So probably these should be marked as "*Possibly* non-namespaced translation key" rather than "Probably non-namespaced translation key" if we want to keep them. And I would argue that these occurrences can still be helpful when trying to properly namespace all keys for use in `i18n/label`, i.e. when solving #17811. There is additional code to list just the "Probably non-namespaced translation keys" themselves, without showing where they are used in `src`. This can be a starting point to solve #17811, but would basically require the developer to grep for those keywords in `src` themself (which is why I wanted in addition to that also list all the individual occurrences in the first place).

Once #17811 and also #17813 are solved, I would argue that there is little benefit of keeping this functionality (finding probably non-namespaced translation keys and listing all their possible occurrences), since instead we could simply ignore all non-namespaced keywords in `src` altogether, and then forgetting to use `:t/` in `src` for a translation key in `translations/en.json` would result in a "(Probably) Unused translation key" warning which should be enough for the developer to go by to resolve it.

So instead, to make the lengthy output more useful, I will sort the verbose output of possibly non-namespaced translation keys by the translation keys, rather than by the files (as is currently happening), since that is how that would be used by a developer solving #17811.

> I believe this should be fixed for the script to be ready for merge or the part of the code that produces that output removed because it's unusable as it is now.

I would argue to keep the functionality around for now, specifically to help resolve #17811. And then remove it after #17813 is resolved.
The alternative is to wait with #17811 and e.g. to rely on `malli` for catching non-namespaced usages of translation keys, but we may not catch them all this way.

> On the other hand, `Probably unused translation key` messages are much better, those really seem to not be used as far as I tested and are actually useful.

Compared to non-namespaced translation keys this is of course the low-hanging fruit.

akatov committed:
control individual functionality using flag fariables. Sort occurrences of possibly non-namespaced translation keywords by tranlation key.

akatov commented:
@ilmotta , actually, most of the non-namespaced translation keys seem to live in `src/status_im2/setup/i18n_test.cljs`.

There are only 41 other non-namespaced keys left outside of that file...

I'll make a separate PR to namespace the keys in `i18n-test`.

flexsurfer commented:
feel free to remove this test its not used

akatov commented:
> feel free to remove this test its not used

@flexsurfer - I opened #17816 specifically for this.

akatov commented on translations/en.json:
```diff
@@ -646,6 +645,7 @@
     "get-stickers": "Get Stickers",
     "gif": "GIF",
     "go-to-settings": "Go to Settings...",
+    "goerli-network": "Goerli network",
```
See now #17818

akatov commented on src/status_im/ui/screens/ens/views.cljs:
```diff
@@ -147,7 +147,7 @@
     [help-message-text-element
      :t/ens-username-owned
-     :t/ens-username-continue]
+     :t/ens-username-owned-continue]
```
See now #17819

akatov commented:
Closing in favour of #17817, #17818, #17819, and #17820
